scripts/convert_tags_to_custom_properties/convert_tags_to_custom_properties.py

```diff
@@ -0,0 +1,206 @@
+        has_next_page = page_info["hasNextPage"]
+        cursor = page_info["endCursor"]
+
+    return properties
+
+
+def main():
+    properties = fetch_custom_properties()
+
+    for index, property_info in enumerate(properties, 1):
+        property_name = property_info["name"]
+        property_alias = property_info["aliases"][0]  # Only the first alias
+        property_schema_type = property_info["schema"]["type"]
+        print(f"{index}. Property Name: {property_name}, Alias: {property_alias}, Schema Type: {property_schema_type}")
+
+    selected_index = int(input("Select a property by entering its index: ")) - 1
+    if 0 <= selected_index < len(properties):
+        selected_property = properties[selected_index]
+        print(f"You selected: {selected_property['name']} with alias: {selected_property['aliases'][0]} and schema type: {selected_property['schema']['type']}")
+
+        service_mutations = {
+            "boolean": execute_boolean_mutation,
+            "array": execute_array_mutation,
+            "text": execute_text_mutation,
+            "object": execute_object_mutation
+        }
+
+        mutation_executor = service_mutations.get(selected_property["schema"]["type"])
+        if mutation_executor:
+            mutation_executor(selected_property)
+        else:
+            print("Unsupported schema type.")
+
+
+def execute_boolean_mutation(property_info):
+    """
+    Execute mutation for boolean schema type.
+    """
+    cursor = None
+    while True:
+        # Execute the GraphQL query using the selected alias as the tag key
+        response_services_by_tag = opslevel_graphql_query(
+            SERVICES_BY_TAG_QUERY, variables={"endCursor": cursor, "tag_key": property_info["aliases"][0]}
+        )
+
+        # Loop through the service nodes to confirm the key in tags nodes matches the selected alias
+        services = response_services_by_tag["data"]["account"]["services"]["nodes"]
+        for service in services:
+            tags = service["tags"]["nodes"]
+            for tag in tags:
+                if tag["key"] == property_info["aliases"][0]:
+                    print(f"Service ID: {service['id']} has the selected alias as a tag.")
+                    # Execute the mutation query
+                    response = opslevel_graphql_query(
+                        UPDATE_PROPERTY_MUTATION,
+                        variables={"service_id": service["id"], "definition_alias": property_info["aliases"][0], "value": tag["value"]}
+                    )
+                    # # Process the mutation response if needed
+                    print("Bool mutation executed.")
+                    print(response)
+        else:
+            print("No service found with the selected alias as a tag.")
+
+        # Check if there are more pages
+        if not response_services_by_tag["data"]["account"]["services"]["pageInfo"]["hasNextPage"]:
```

Comment:
Recommend saving this to a variable and using it as the condition for the `while` to increase readability

Reply:
made improvement based on suggestion.